### PR TITLE
Prem SDK integration in LangChain

### PR DESCRIPTION
**Pull request by Anindyadeep**
### Prem SDK integration in LangChain

This PR adds the integration with [PremAI's](https://www.premai.io/) prem-sdk with llama-index. User can now access to deployed models (llms/embeddings) and use it with langchain's ecosystem. This PR adds the following:

### Todo:

- [x]  Add chat support
- [X]  Adding embedding support
- [X]  writing integration tests
    - [X]  writing tests for chat 
    - [X]  writing tests for embedding
- [X]  writing unit tests
    - [X]  writing tests for chat 
    - [X]  writing tests for embedding
- [X]  Adding documentation
    - [X]  writing documentation for chat
    - [X]  writing documentation for embedding
- [X] run `make test`
- [X] run `make lint`, `make lint_diff` 
- [X]  Final checks (spell check, lint, format and overall testing)

### CODE REVIEW

**Review by Janaka-Steph — COMMENTED** [review]
I think that chat completion should extends `BaseChatModel`or `SimpleChatModel` instead of `BaseLLM`

**Comment by Anindyadeep** [comment]
> I think that chat completion should extends `BaseChatModel`or `SimpleChatModel` instead of `BaseLLM`

Yes, but this was for LLM integration not the chat model. I will do the chat_model implementation too.

**Comment by Anindyadeep** [comment]
Although I have some items left, but since code part is completed, so an earlier review and feedbacks would be awesome to change on the fly :) 

cc: @filopedraz @Janaka-Steph

**Review by filopedraz — APPROVED** [review]

**Comment by filopedraz** [comment]
Seems great

**Comment by Anindyadeep** [comment]
> Seems great

Just added documentation for Embeddings :)

File: libs/community/langchain_community/chat_models/prem.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    """Use any LLM provider with Prem and Langchain.
+
+    To use, you will need to have an API key. You can find your existing API Key
+    or generate a new one here: https://app.premai.io/api_keys/
+    """
+    
+    # TODO: Discussion needed. We can any one the following:
+    
+    # - User can set/override parameters here and then we can set it in additional arguments
+    # - Remove all the optional paramter here and keep it to the kwargs (but that would avoid function defintion)
```
[the remark this answers is not in the document]
Comment (filopedraz) [reply libs/community/langchain_community/chat_models/prem.py]: @giowe

**Pull request merged**